components/async-status-mgt/org.wso2.carbon.identity.framework.async.status.mgt/src/main/java/org/wso2/carbon/identity/framework/async/status/mgt/dao/AsyncStatusMgtDAOImpl.java

```diff
@@ -0,0 +1,880 @@
+
+        List<ResponseOperationRecord> responseContexts = new ArrayList<>();
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, operationSubjectType);
+            statement.setString(2, operationSubjectId);
+            statement.setString(3, operationType);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                while (resultSet.next()) {
+                    ResponseOperationRecord responseContext = new ResponseOperationRecord();
+                    responseContext.setOperationId(resultSet.getString(
+                            IDN_OPERATION_ID));
+                    responseContext.setCorrelationId(resultSet.getString(
+                            SQLConstants.OperationStatusTableColumns.IDN_CORRELATION_ID));
+                    responseContext.setOperationType(
+                            resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_TYPE));
+                    responseContext.setOperationSubjectType(
+                            resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_SUBJECT_TYPE));
+                    responseContext.setOperationSubjectId(
+                            resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_SUBJECT_ID));
+                    responseContext.setResidentOrgId(
+                            resultSet.getString(
+                                    SQLConstants.OperationStatusTableColumns.IDN_OPERATION_INITIATED_ORG_ID));
+                    responseContext.setInitiatorId(
+                            resultSet.getString(
+                                    SQLConstants.OperationStatusTableColumns.IDN_OPERATION_INITIATED_USER_ID));
+                    responseContext.setOperationStatus(
+                            resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_STATUS));
+                    responseContext.setOperationPolicy(
+                            resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_POLICY));
+                    responseContext.setCreatedTime(
+                            Timestamp.valueOf(
+                                    resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_CREATED_AT)));
+                    responseContext.setModifiedTime(
+                            Timestamp.valueOf(
+                                    resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_LAST_MODIFIED)));
+                    responseContexts.add(responseContext);
+                }
+            }
+        } catch (SQLException e) {
+            String errorMessage = "Error fetching async operation status for subject: " + operationSubjectId;
+            LOGGER.info(errorMessage + e);
```

Comment:
```suggestion
            LOGGER.error(errorMessage + e);
```

Shouldn't this be an error ?